src/redux/data.ts

```diff
@@ -825,45 +832,65 @@ export const maybeFetchF2CHashForPendingTransactions = async (
             `maybeFetchF2CHashForPendingTransactions: fetching order`
           );
 
-          const { data, error } = await gretch<ActivityItem>(
-            `https://f2c.rainbow.me/v1/providers/ratio/users/${userId}/activity/${orderId}`
-          ).json();
+          try {
+            const data = await queryClient.fetchQuery({
+              queryKey: ['f2c', 'ratio', 'pending_tx_check'],
+              staleTime: 10_000, // only fetch AT MOST once every 10 seconds
+              async queryFn() {
+                const { data, error } = await gretch<ActivityItem>(
+                  `https://f2c.rainbow.me/v1/providers/ratio/users/${userId}/activity/${orderId}`
+                ).json();
+
+                if (!data || error) {
+                  if (error) {
+                    throw error;
+                  } else {
+                    throw new Error(
+                      'Ratio API returned no data for this transaction'
+                    );
+                  }
+                }
+
+                return data;
+              },
+            });
 
-          loggr.debug(
-            `maybeFetchF2CHashForPendingTransactions: fetched order`,
-            {
-              hasData: Boolean(data),
-              hasError: Boolean(error),
-              hasHash: Boolean(data?.crypto?.transactionHash),
-            }
-          );
+            loggr.debug(
```

Comment:
Same handling from here down, just inside a try/catch because `fetchQuery` throws if the `queryFn` throws.